README.md

```diff
@@ -2,34 +2,47 @@
+check a special service or condition. For this, the plugin is called with the
+option *check*. If this fails, the plugin will exit with the return value `1`,
+else `0`. Have a look at the default plugins shipped in
+`/usr/local/libexec/health-checker` for examples.
+
+The amount of times a new snapshot is rebooted before considering
+it as not working is defined in `/etc/kernel/tries` (_3_ by default).
+
+`health-checker` reboot policy can also be configured by passing the following values
+as kernel arguments:
+
+- `health-checker-reboot=force`, force health-checker to reboot the system if
+  there are tries left for the current snapshot. By default, the system gets rebooted
+  if the boot entry is the default one (usually the newest one).
+- `health-checker-reboot=disabled`, disable the reboot when `health-checker` fails.
 
 ## How does this work?
 
-`health-checker` will be called by a systemd service during the boot
-process. All services, which should be checked, needs to be listed in the
-'After' section.
-
-The `health-checker` script will call several plugins. Every plugin is
-responsible to check a special service or condition. For this, the plugin is
-called with the option *check*. If this fails, the plugin will exit with the
-return value `1`, else `0`.
-If everyting was fine, the script will create a
-`/var/lib/misc/health-check.state` file with the number of the current,
-working btrfs subvolume with the root filesystem.
-If a plugin reports an error condition, the `health-checker` script will take
-following actions:
-
-1. If the current btrfs root subvolume is not identical with the last known
-   working snapshot, an automatic rollback to that snapshot is made. Normally,
-   if the current btrfs subvolume is not identical to the last working one,
-   this means an update was made, and this update did never boot correctly.
-2. If the current btrfs subvolume did already boot successful in the past, the
-   problem is most likely a temporary problem. In this case, we try to reboot
-   the machine again. `/var/lib/misc/health-check.rebooted` will be created
-   with the current time.
-3. If the current btrfs snapshot did already boot successful in the past and
-   if we did try already to solve the problem with a reboot, it doesn't make
-   sense to reboot again. To give the admin the chance and possibility to fix
-   the problem, all plugins will be called with the option *stop*. At the
-   end, the machine should still run, so that an admin can login, but no
-   service should run, so that nothing can break.
+`health-checker` assess the status of the system at the boot and act accordingly.
+It uses the [Automatic Boot Assessment](https://systemd.io/AUTOMATIC_BOOT_ASSESSMENT/)
+provided by systemd to check every new snapshot, to reboot a predetermined amount
+of times, and then it lets the user access an emergency shell when everything
+else fails.
+
+Every new snapshot has a separate boot entry with a boot counter (according to
```

Comment:
"Every new snapshot" or just "Every snapshot"?

Reply:
The boot count is enabled by `kernel-install` when `/etc/kernel/tries` is set, so this only works for new snapshot. `systemd-bless-boot` will reset the counter when the new snapshot boots successfully, and it is never enabled again.